src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -40,6 +40,19 @@
 #include "memory/metaspaceUtils.hpp"
 #include "memory/metaspaceStats.hpp"
 
+// Regions cannot be uncommitted when concurrent reset is zeroing out the bitmaps.
+// This CADR class enforces this by forbidding region uncommits while it is in scope.
+struct ShenandoahForbidRegionUncommit : public StackObj {
```

Comment:
This is `class`, not `struct`, right? A common name for these in Hotspot are `*No*Mark`, so `ShenandoahNoUncommitMark`?